qiskit_nature/problems/second_quantization/lattice/lattice/triangular_lattice.py

```diff
@@ -0,0 +1,198 @@
+        for x, y in coordinates:
+            node_a = y * rows + x
+            for i in range(3):
+                # x direction
+                if i == 0 and x != rows - 1:
+                    node_b = node_a + 1
+                # y direction
+                elif i == 1 and y != cols - 1:
+                    node_b = node_a + rows
+                # diagonal direction
+                elif i == 2 and x != rows - 1 and y != cols - 1:
+                    node_b = node_a + 1 + rows
+                else:
+                    continue
+                graph.add_edge(node_a, node_b, edge_parameter[i])
+
+        # add self-loops
+        for x, y in coordinates:
+            node_a = y * rows + x
+            graph.add_edge(node_a, node_a, onsite_parameter)
+
+        # depend on boundary condition
+        self.boundary_edges = []
+        # add edges when the boundary condition is periodic.
+        if boundary_condition == "periodic":
+            # The periodic boundary condition in the x direction.
+            # It makes sense only when rows is greater than 2.
+            if rows > 2:
+                for y in range(cols):
+                    node_a = (y + 1) * rows - 1
+                    node_b = node_a - (rows - 1)  # node_b < node_a
+                    graph.add_edge(node_b, node_a, edge_parameter[0].conjugate())
+                    self.boundary_edges.append((node_b, node_a))
+            # The periodic boundary condition in the y direction.
+            # It makes sense only when cols is greater than 2.
+            if cols > 2:
+                for x in range(rows):
+                    node_a = rows * (cols - 1) + x
+                    node_b = x  # node_b < node_a
+                    graph.add_edge(node_b, node_a, edge_parameter[1].conjugate())
+                    self.boundary_edges.append((node_b, node_a))
+            # The periodic boundary condition in the diagonal direction.
+            for y in range(cols - 1):
+                node_a = (y + 1) * rows - 1
+                node_b = node_a + 1  # node_b > node_a
+                graph.add_edge(node_a, node_b, edge_parameter[2])
+                self.boundary_edges.append((node_b, node_a))
+
+            for x in range(rows - 1):
+                node_a = rows * (cols - 1) + x
+                node_b = x + 1  # node_b < node_a
+                graph.add_edge(node_b, node_a, edge_parameter[2].conjugate())
+                self.boundary_edges.append((node_b, node_a))
+
+            node_a = rows * cols - 1
+            node_b = 0  # node_b < node_a
+            graph.add_edge(node_b, node_a, edge_parameter[2].conjugate())
+            self.boundary_edges.append((node_a, node_b))
+
+        elif boundary_condition != "open":
+            raise ValueError(
+                f"Invalid `boundary condition` {boundary_condition} is given."
+                "`boundary condition` must be `open` or `periodic`."
+            )
```

Comment:
Thanks for the suggestion. As I mentioned above, it will be done when we need to make a more general class, but it seems that we don't need such an extraction right now.

Reply:
What @dlasecki was suggesting here (and above) is not necessarily to extract this into a more general class but to extract the various code blocks into separate functions (within this class). This helps with readability because you don't have one long function but rather call out to other functions to do self-contained pieces of work.
It will also be beneficial in the future when doing refactoring because the code is already split into smaller more manageable pieces.

Reply:
Thanks for the comment. As in the class `HyperCubicLattice`, I extracted some code blocks as private functions here, too.